include/bluetooth/mesh/main.h

```diff
@@ -425,14 +425,108 @@ int bt_mesh_lpn_set(bool enable);
  */
 int bt_mesh_lpn_poll(void);
 
-/** @brief Register a callback for Friendship changes.
+/** Low Power Node callback functions. */
+struct bt_mesh_lpn_cb {
+	/** @brief Friendship established.
+	 *
+	 *  This callback notifies the application that friendship has
+	 *  been successfully established.
+	 *
+	 *  @param net_idx  NetKeyIndex used during established friendship.
+	 *  @param friend_addr Friend address.
+	 *  @param queue_size  Friend queue size.
+	 *  @param recv_window Low Power Node's listens duration for
+	 *  Friend response.
+	 */
+	void (*established)(uint16_t net_idx, uint16_t friend_addr,
+			    uint16_t queue_size, uint8_t recv_window);
+
+	/** @brief Friendship terminated.
+	 *
+	 *  This callback notifies the application that friendship has
+	 *  been terminated.
+	 *
+	 *  @param net_idx  NetKeyIndex used during established friendship.
+	 *  @param friend_addr Friend address.
+	 */
+	void (*terminated)(uint16_t net_idx, uint16_t friend_addr);
+
+	/** @brief Local Poll Request.
+	 *
+	 *  This callback notifies the application that the local node has
+	 *  polled the friend node.
+	 *
+	 *  This callback will be called before @ref bt_mesh_lpn_cb::established
+	 *  when attempting to establish a friendship.
+	 *
+	 *  @param net_idx  NetKeyIndex used during established friendship.
+	 *  @param friend_addr Friend address.
+	 *  @param retry Retry or first poll request for each transaction.
+	 */
+	void (*polled)(uint16_t net_idx, uint16_t friend_addr, bool retry);
+
+	/* Internal linked list node. */
+	sys_snode_t node;
+};
+
+/** @brief Register a callback structure for Friendship events.
  *
- *  Registers a callback that will be called whenever Friendship gets
- *  established or is lost.
+ *  Registers a callback structure that will be called whenever Friendship
+ *  gets established or terminated, or a poll request is sent.
  *
- *  @param cb Function to call when the Friendship status changes.
+ *  @param cb Function structure to call when the Friendship events changes.
+ */
+void bt_mesh_lpn_register_cb(struct bt_mesh_lpn_cb *cb);
+
+/** Friend Node callback functions. */
+struct bt_mesh_friend_cb {
+	/** @brief Friendship established.
+	 *
+	 *  This callback notifies the application that friendship has
+	 *  been successfully established.
+	 *
+	 *  @param net_idx  NetKeyIndex used during established friendship.
+	 *  @param lpn_addr Low Power Node address.
+	 *  @param recv_delay Receive Delay in unites of 1 millisecond.
+	 *  @param polltimeout PollTimeout in units of 1 millisecond.
+	 */
+	void (*established)(uint16_t net_idx, uint16_t lpn_addr,
+			    uint16_t recv_delay, int32_t polltimeout);
```

Comment:
Why is `polltimeout` `int32_t` and not `uint32_t`? Also, why is `recv_delay` `uint16_t` when it's only one octet in Friend Requests?